src/trace-writer.ts

```diff
@@ -283,7 +264,21 @@ export class TraceWriter extends common.Service {
             'TraceWriter#queueTrace: Trace buffer full, flushing.');
         setImmediate(() => this.flushBuffer());
       }
-    });
+    };
+    // TODO(kjin): We should always be following the 'else' path.
+    // Any test that doesn't mock the Trace Writer will assume that traces get
+    // buffered synchronously. We need to refactor those tests to remove that
+    // assumption before we can make this fix.
+    if (this.config.projectId) {
+      afterProjectId(this.config.projectId);
```

Comment:
Maybe add a warning log here so that if this code path is executed in production, people can notice?

Reply:
The unstated corollary to `We should always be following the 'else' path.` is `we are currently almost always following the 'if' path`. AFAIK there should be nothing user-visibly different from before; this TODO just comments on the unintuitive reason for branching.

Reply:
LOL. I thought you meant that `if` path was executed only from tests. Is this comment useful? I was easily confused.

Reply:
It's basically a long way of saying "I don't like how it's written right now, I want to change it later" 😅